src/lib/gssapi/krb5/accept_sec_context.c

```diff
@@ -782,6 +783,7 @@ kg_accept_krb5(minor_status, context_handle,
                 goto fail;
             }
 
+            have_cb_success = 1;
```

Comment:
On a closer look, I see two problems:

* The new code doesn't implement bullet point 4 of section 3 of the draft.  If the acceptor caller supports channel bindings (communicated via ret_flags_understood), the acceptor provided channel bindings, and the initiator didn't supply any (all bytes of the wire checksum are 0), then we shouldn't fail out; we should instead just not set the channel-bound flag in ret_flags.

* The existing code doesn't implement bullet point 2 of section 3 of the draft when ``authdat->checksum == NULL`` or when ``authdat->checksum->checksum_type != CKSUMTYPE_KG_CB``.

Also, there is significant technical debt present in the old code: the checksum processing code is open-coded into a function which is much too long even without it.  Because I don't like adding to existing technical debt, I'm going to insist on factoring out the checksum processing code before making it more complicated with the channel-bound flag.  That refactoring should take place in a second commit and shouldn't affect behavior.